crates/sage-cloudpath/src/tdf.rs

```diff
@@ -0,0 +1,61 @@
+use rayon::prelude::*;
+use sage_core::spectrum::{Precursor, RawSpectrum, Representation};
+use std::fmt::{Display, Formatter};
+use timsrust;
+
+#[derive(Default)]
+pub struct TdfReader {}
+
+impl TdfReader {
+    pub fn parse(
+        &self,
+        path_name: impl AsRef<str>,
+        file_id: usize,
+    ) -> Result<Vec<RawSpectrum>, TdfError> {
+        let dda_spectra: Vec<timsrust::Spectrum> =
+            timsrust::FileReader::new(path_name.as_ref().to_string()).read_all_spectra();
+        let spectra: Vec<RawSpectrum> = (0..dda_spectra.len())
+            .into_par_iter()
+            .map(|index| {
+                let dda_spectrum = &dda_spectra[index];
+                let mut precursor: Precursor = Precursor::default();
+                let dda_precursor: timsrust::Precursor =
+                    dda_spectrum.precursor.unwrap_as_precursor();
+                precursor.mz = dda_precursor.mz as f32;
+                precursor.charge = Option::from(dda_precursor.charge as u8);
+                // precursor.ion_mobility = Option::from(dda_precursor.im as f32);
+                precursor.intensity = Option::from(dda_precursor.intensity as f32);
+                precursor.spectrum_ref = Option::from(dda_precursor.frame_index.to_string());
+                let spectrum: RawSpectrum = RawSpectrum {
+                    file_id: file_id,
+                    precursors: vec![precursor],
+                    representation: Representation::Centroid,
+                    scan_start_time: dda_precursor.rt as f32,
+                    ion_injection_time: dda_precursor.rt as f32,
+                    total_ion_current: 0.0,
+                    mz: dda_spectrum.mz_values.iter().map(|&x| x as f32).collect(),
+                    ms_level: 2,
```

Comment:
You can do this with the `read_all_frames` function of the `FileReader`. You would then do a filtering to retain only the ms1 frames. Would it make sense to provide a separate function for that? Could also be useful for performance reasons. MS1 frames don't really need to be read for MS2 spectra and vice verse MS2 frames have no use for MS1 frames. Current implementation loads both twice...

Reply:
Some kind of function, or description of how to do so would be great - once I have a frame, do I need to perform some more operations to reconstruct a spectra (not super familiar with Bruker data)?

Reply:
I'll put a separate function in there. That said, incorporating TIMSTOF data in sage for LFQ is non-trivial, as the ion mobility dimension means a frame is not directly convertible to a regular mz spectrum. We cna do a projection on the mz axis, but a lot of info will be missed like this (might be an acceptable firt approach though, if only to make it possible without good performance)

Reply:
Yeah, it will eventually require a rewrite of the LFQ module. I think a projection onto the m/z access sounds like a good first pass, at least to get some minimum functionality in place.

Reply:
Hey there! sorry for pitching in but I think a simple addition that can increase the S/N in some of my experiments is to first filter the frame for the mass ranges that actually get fragmented. In the sketch I am adding, the green area is the section that usually gets targeted for fragmentation, therefore anything outside of that has very little value for LFQ (and in my experience adds a lot of noise), so keeping only the intensities/mz_values that are within that area can be a good idea prior to projecting the data onto the m/z axis.

![image](https://github.com/lazear/sage/assets/8116459/a461fc0c-eec7-4cee-ac14-46cc29f05a5b)

Reply:
@jspaezp this is great, thanks - also, ready for the IM stuff you did if you want to open a PR